src/components/wrapper/song-list-wrapper.tsx

```diff
@@ -0,0 +1,158 @@
+import { ChangeEvent, useEffect, useRef, useState } from "react"
+import { PlusIcon } from "@heroicons/react/24/solid"
+import { useSupabaseClient } from "@supabase/auth-helpers-react"
+import { Database } from "@/types/database"
+import { Suggestion } from "@/types/database-types"
+import { MagnifyingGlassCircleIcon } from "@heroicons/react/24/outline"
+import Spinner from "@/components/utils/spinner"
+import ErrorPopup from "@/components/popups/error-popup"
+import { useRouter } from "next/router"
+import { MagnifyingGlassIcon } from "@heroicons/react/20/solid"
+import SearchBar from "@/components/suggestion/search-bar"
+import { getRepertoireSongs, getSuggestions } from "@/services/suggestion.service"
+
+interface SongListPageWrapperProps {
+  renderSongCard: (suggestion: Suggestion) => JSX.Element
+  pageName: string
+}
+
+const SongListPageWrapper = (props: SongListPageWrapperProps) => {
```

Comment:
Since the name of the file is SongListWrapper I'd recommend also renaming and its props to SongListWrapper.